test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1

```diff
@@ -16,10 +16,24 @@ Describe "Basic Auth over HTTP not allowed on Unix" -Tag @("CI") {
         $err = ({New-PSSession -ComputerName 'localhost' -Credential $credential -Authentication Basic}  | Should -Throw -PassThru  -ErrorId 'System.Management.Automation.Remoting.PSRemotingDataStructureException,Microsoft.PowerShell.Commands.NewPSSessionCommand')
         $err.Exception | Should -BeOfType [System.Management.Automation.Remoting.PSRemotingTransportException]
         # Should be PSRemotingErrorId.ConnectFailed
-        # Ensures we are looking at teh expected instance
+        # Ensures we are looking at the expected instance
         $err.Exception.ErrorCode | Should -Be 801
     }
 
+   # Marked as pending due tohttps://github.com/Microsoft/omi/issues/502
+   # It "New-PSSession should NOT throw a ConnectFailed exception when specifying Basic Auth over HTTPS on Unix" -skip:($IsWindows) {
+    It "New-PSSession should NOT throw a ConnectFailed exception when specifying Basic Auth over HTTPS on Unix" -Pending {
+        $password = ConvertTo-SecureString -String "password" -AsPlainText -Force
+        $credential = [PSCredential]::new('username', $password)
+
+        # use a Uri that specifies HTTPS to test Basic Auth logic.
+        # NOTE: The connection is expected to fail but not with a  ConnectFailed exception
+        $uri = "https://localhost"
+        New-PSSession -Uri $uri -Credential $credential -Authentication Basic -ErrorVariable err
```

Comment:
Can you use the pattern below?

```PowerShell
{ New-PSSession -Uri $uri -Credential $credential -Authentication Basic -ErrorVariable err } | Should -Throw -ErrorId '1,PSSessionOpenFailed' -ExceptionType 'System.Management.Automation.Remoting.PSRemotingTransportException'
$err.Exception.HResult | Should -Be -2146233087
```

Reply:
No, when -ErrorVariable is used, pester doesn't have an exception to catch.

Reply:
This cannot be done as when `err` is set no exception is thrown.

Reply:
If this is a negative test case where you need to do more validation of the exception, you should use the pattern in our test guidance doc with `-PassThru`: https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/WritingPesterTests.md#writing-pester-tests

Reply:
@SteveL-MSFT: new-pssession doesn't throw, it writes to the error stream.

Reply:
@SteveL-MSFT: do you have any additional feedback?

Reply:
I think you should follow the convention for now to use `-ErrorAction Stop` and then use Pester's `Should -ThrowErrorId -PassThru` to do additional validation for consistency until we have capability to validate non-terminating errors